# HG changeset patch
# User Mike Miller <mtmiller@ieee.org>
# Date 1424554621 18000
#      Sat Feb 21 16:37:01 2015 -0500
# Node ID 7dbb8aa43d56e626b0ff1daf22fc585dbcfca9b3
# Parent  71770cf07c30fd3817e3e8980364d9e207cb33a8
Install correct freedesktop AppData file contents (bug #43279)

* etc/icons/octave.appdata.xml.in: Move here from etc and add
%OCTAVE_DESKTOP_FILE% substitution pattern.
* etc/icons/Makefile.am (DESKTOP_FILE, VENDOR): New variables.
(octave.appdata.xml): New rule.  (install-data-local, uninstall-local):
Install and uninstall octave.appdata.xml.  (EXTRA_DIST): Include
octave.appdata.xml.in in the list.  (CLEANFILES): Include octave.appdata.xml
in the list.
* Makefile.am (EXTRA_DIST): Remove etc/octave.appdata.xml from the list.

diff --git a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -58,17 +58,16 @@ EXTRA_DIST = \
   etc/README.Cygwin \
   etc/README.Linux \
   etc/README.MacOS \
   etc/README.MinGW \
   etc/README.Windows \
   etc/README.gnuplot \
   etc/README.kpathsea \
   etc/gdbinit \
-  etc/octave.appdata.xml \
   run-octave.in
 
 include m4/module.mk
 
 # Subdirectories in which to run `make all'.  Including "." before
 # @DOCDIR@ is an attempt to force all preceding directories in the list
 # to be processed before the current directory so that the
 # scripts/DOCSTRINGS libinterp/DOCSTRINGS files are built before
diff --git a/etc/icons/Makefile.am b/etc/icons/Makefile.am
--- a/etc/icons/Makefile.am
+++ b/etc/icons/Makefile.am
@@ -42,17 +42,27 @@ WINDOWS_PNG_ICONS = $(filter %-16.png %-
 BUILT_ICONS = \
   $(BUILT_PNG_ICONS) \
   octave-logo.ico
 
 image_DATA = \
   $(IMAGE_FILES) \
   octave-logo.ico
 
-all-local: octave.desktop $(BUILT_ICONS)
+VENDOR = www.octave.org
+
+DESKTOP_FILE = $(VENDOR)-octave.desktop
+
+all-local: octave.appdata.xml octave.desktop $(BUILT_ICONS)
+
+octave.appdata.xml: octave.appdata.xml.in Makefile
+	$(AM_V_GEN)rm -f $@-t $@ && \
+	$(SED) < $< > $@-t \
+	  -e "s|%OCTAVE_DESKTOP_FILE%|${DESKTOP_FILE}|" && \
+	mv $@-t $@
 
 octave.desktop: octave.desktop.in Makefile
 	$(AM_V_GEN)rm -f $@-t $@ && \
 	$(SED) < $< > $@-t \
 	  -e "s|%OCTAVE_IMAGEDIR%|${imagedir}|" \
 	  -e "s|%OCTAVE_PREFIX%|${prefix}|" && \
 	mv $@-t $@
 
@@ -64,44 +74,48 @@ octave.desktop: octave.desktop.in Makefi
 octave-logo.ico: $(WINDOWS_PNG_ICONS)
 	$(AM_V_GEN)rm -f $@-t $@ && \
 	$(ICOTOOL) --create --raw $^ > $@-t && \
 	mv $@-t $@
 
 install-data-local:
 	-if test -n "$(DESKTOP_FILE_INSTALL)"; then \
 	  $(DESKTOP_FILE_INSTALL) --dir=$(DESTDIR)$(datadir)/applications \
-	    --vendor www.octave.org octave.desktop; \
+	    --vendor $(VENDOR) octave.desktop; \
 	fi
 	for f in $(BUILT_PNG_ICONS); do \
 	  size=`echo $$f | $(SED) -n -e "s/.*-\([0-9]\+\)\.png/\1/p"`; \
 	  if test -f $$f; then d=; else d="$(srcdir)/"; fi; \
 	  $(MKDIR_P) $(DESTDIR)$(datadir)/icons/hicolor/$${size}x$${size}/apps; \
 	  $(INSTALL_DATA) "$$d$$f" $(DESTDIR)$(datadir)/icons/hicolor/$${size}x$${size}/apps/octave.png; \
 	done
 	$(MKDIR_P) $(DESTDIR)$(datadir)/icons/hicolor/scalable/apps
 	$(INSTALL_DATA) $(srcdir)/octave-logo.svg $(DESTDIR)$(datadir)/icons/hicolor/scalable/apps/octave.svg
+	$(MKDIR_P) $(DESTDIR)$(datadir)/appdata
+	$(INSTALL_DATA) octave.appdata.xml $(DESTDIR)$(datadir)/appdata/$(VENDOR)-octave.appdata.xml
 
 uninstall-local:
 	if test -n "$(DESKTOP_FILE_INSTALL)"; then \
-	  rm -f $(DESTDIR)$(datadir)/applications/www.octave.org-octave.desktop; \
+	  rm -f $(DESTDIR)$(datadir)/applications/$(DESKTOP_FILE); \
 	fi
 	for f in $(BUILT_PNG_ICONS); do \
 	  size=`echo $$f | $(SED) -n -e "s/.*-\([0-9]\+\)\.png/\1/p"`; \
 	  rm -f $(DESTDIR)$(datadir)/icons/hicolor/$${size}x$${size}/apps/octave.png; \
 	done
 	rm -f $(DESTDIR)$(datadir)/icons/hicolor/scalable/apps/octave.svg
+	rm -f $(DESTDIR)$(datadir)/appdata/$(VENDOR)-octave.appdata.xml
 
 if AMCOND_HAVE_ICON_TOOLS
 else
 dist-hook:
 	@echo "Packaging distribution requires icotool and rsvg-convert." ; exit 1;
 endif
 
 EXTRA_DIST = \
   $(BUILT_ICONS) \
   $(IMAGE_FILES) \
+  octave.appdata.xml.in \
   octave.desktop.in
 
-CLEANFILES = octave.desktop
+CLEANFILES = octave.appdata.xml octave.desktop
 
 MAINTAINERCLEANFILES = $(BUILT_ICONS)
 
diff --git a/etc/octave.appdata.xml b/etc/icons/octave.appdata.xml.in
rename from etc/octave.appdata.xml
rename to etc/icons/octave.appdata.xml.in
--- a/etc/octave.appdata.xml
+++ b/etc/icons/octave.appdata.xml.in
@@ -11,17 +11,17 @@
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
   GNU General Public License for more details.
 
   You should have received a copy of the GNU General Public License
   along with this program.  If not, see <http://www.gnu.org/licenses/>.
 -->
 <application>
-  <id type="desktop">octave.desktop</id>
+  <id type="desktop">%OCTAVE_DESKTOP_FILE%</id>
   <licence>GPLv3+</licence>
   <name>GNU Octave</name>
   <summary>Interactive programming environment for numerical computations</summary>
   <description>
     <p>
       GNU Octave is a high-level interpreted language, primarily intended for
       numerical computations. It provides capabilities for the numerical
       solution of linear and nonlinear problems, and for performing other
